tests/functionality_tests/test_activation_shapes.py

```diff
@@ -18,13 +17,26 @@ def main():
         print("flip rows, cols for traversal")
 
     activation_type = input("Activation type: ")
-    is_vertical = "True" == input("Vertical traversal: ")
-    if is_vertical:
+    direction = input("Traversal Direction: ")
+    if direction.isnumeric():
+        numberToDirection = {"0": "RIGHT", "1": "UP", "2": "LEFT", "3": "DOWN"}
+        direction = numberToDirection.get(direction)
+    else:
+        direction = direction.upper()
+
+    print(direction)
+    if direction == "UP":
         grid.direction = grid.Direction.UP
+    elif direction == "DOWN":
+        grid.direction = grid.Direction.DOWN
+    elif direction == "RIGHT":
+        grid.direction = grid.Direction.RIGHT
+    else:
+        grid.direction = grid.Direction.LEFT
```

Comment:
i think else condition should be elif and add raise error for else

Reply:
Done